modules/ROOT/pages/troubleshoot-adapter-provisioning-issues.adoc

```diff
@@ -0,0 +1,180 @@
+== Mule Runtime Engine Fails to Start Due to Licence Issues
+
+When you attempt to list your adapters, you might receive a failed status.
+
+=== Cause
+
+If you did not provide a valid Mule license with the required entitlements, your Mule adapter will not start.
+
+=== Diagnosis
+
+To diagnose the issue:
+
+. Verify the adapter status:
++
+`$ ./service-mesh adapter list`.
++
+. If the adapter status is displayed as `Failed`, verify the status of the (grpmule-<namespace>) adapter pod:
++
+`$ kubectl -n service-mesh get pods`.
+In this case, your pod status might be displayed as `CrashLoopBackOff`.
+
+=== Solution
+
+To resolve this issue, update the adapter with a valid license:
++
+`$ ./service-mesh management config license --license=/path/to/license.lic`.
+
+
+[[catalog-content-init-failed]]
+== Catalog Content Initialization Failed
+
+When you attempt to provision an adapter, it might fail because the catalog content did not correctly initialize.
+
+=== Cause
+
+The Catalog is a component installed by Anypoint Service Mesh, which is required to provision adapters, create bindings, and create APIs. If the Catalog is not initialized correctly, the adapter fails.
+
+=== Diagnosis
+
+To diagnose the issue:
+
+. Review the adapter logs:
++
+`$ ./service-mesh adapter logs --namespace=<adapter_namespace> --name=<adapter_name>`.
++
+An output similar to the following text is displayed:
++
+`2019-11-26T20:46:35Z   Error provisioning ServiceInstance of ClusterServiceClass (K8S: "4f6e6cf6-ffdd-425f-a2c7-3c9258ad246a" ExternalName: "anypoint-service-mesh-adapter") at ClusterServiceBroker "service-mesh-incluster-broker": Status: 500; ErrorMessage: <nil>; Description: Catalog content init failed; ResponseError: <nil>`
+
+=== Solution
+
+To resolve this issue, run the following command to force Catalog initialization:
++
+`$ kubectl -n service-mesh patch clusterservicebrokers/service-mesh-incluster-broker --type=json -p='[{"op": "replace", "path": "/spec/relistRequests","value":1}]'`.
+
+[[adapter-failed-to-delete]]
+== Adapter Failed to Delete
+
+When you attempt to delete an adapter that you previously created, you might encounter issues.
+
+=== Cause
+
+If the adapter that you want to delete has service bindings associated with it, you cannot delete that adapter.
+
+=== Diagnosis
+
+To diagnose the issue, verify  the adapter status:
++
+`$ ./service-mesh adapter list`.
+
+In the output of the command, if there are bindings associated with the adapter or the service instance, the status is displayed as `DeprovisionBlockedByExistingCredentials`. This means that the service has associated bindings.
+
+
+=== Solution
+
+To resolve this issue, delete all the bindings associated with the adapter. When every binding is successfully deleted, then the adapter will be also deleted. 
+
+For more information, see xref:https://github.com/GoogleCloudPlatform/kubernetes-engine-samples/tree/master/service-catalog#instance-has-remaining-bindings[Kubernetes] help. 
+
+
```

Comment:
Let's talk some time about tightening up this template. I think we can omit some unnecessary repetition.

Reply:
ok sure!